File: Advanced/TemplaterServer/src/Spire.cs
using Spire.Doc;
using System.IO;

namespace TemplaterServer
{
	public class Spire : PdfConverter
	{
		public Stream Convert(Stream template, string extension)
		{
			var doc = new Document();
			doc.LoadFromStream(template, FileFormat.Docx);
			var ms = new MemoryStream();
			doc.SaveToStream(ms, FileFormat.PDF);
			ms.Position = 0;
			return ms;
		}
	}
}

File: Advanced/TemplaterServer/src/main/java/hr/ngs/templater/server/Spire.java
package hr.ngs.templater.server;

import com.spire.doc.Document;

import java.io.*;

public class Spire implements PdfConverter {

	@Override
	public byte[] convert(byte[] template, String extension) {
		ByteArrayInputStream input = new ByteArrayInputStream(template);
		ByteArrayOutputStream result = new ByteArrayOutputStream();
		Document doc = new Document();
		doc.loadFromStream(input, com.spire.doc.FileFormat.Docx);
		doc.saveToFile(result, com.spire.doc.FileFormat.PDF);
		return result.toByteArray();
	}
}
